### PR TITLE
fix(logging): implement scoped logging

### LINKED ISSUE #151: aiohttp backend needs its own logger
https://github.com/Kitware/wslink/blob/9a6243c7029cda26a827590210783b030e49ca01/python/src/wslink/backends/aiohttp/__init__.py#L101

This module is currently using the `root` logger. It should be using its own logger, e.g.:

```py
import logging
logger = logging.getLogger(__name__)
```

### PR DESCRIPTION
**Pull request by banesullivan**
Implements properly scoped logging. Prior to these changes, `wslink` was hijacking the root logger for its logging and setting global configurations that would conflict with settings for other packages or by users.

fix #151

### CODE REVIEW

File: python/src/wslink/backends/aiohttp/launcher.py [hunk 1]
```diff
@@ -213,15 +215,17 @@ def startWebServer(options, config):
...
+    fh.setLevel(logging.DEBUG)
+    fh.setFormatter(logging.Formatter(formatting))
+    logger.addHandler(fh)
     if options.debug:
         console = logging.StreamHandler(sys.stdout)
         console.setLevel(logging.INFO)
         formatter = logging.Formatter(formatting)
         console.setFormatter(formatter)
-        logging.getLogger("").addHandler(console)
+        logger.addHandler(console)
```
Comment (banesullivan) [inline python/src/wslink/backends/aiohttp/launcher.py:228]:
```suggestion
        logging.getLogger("wslink").addHandler(console)
```

File: python/src/wslink/backends/aiohttp/launcher.py [hunk 2]
```diff
@@ -213,15 +215,17 @@ def startWebServer(options, config):
...
     logFileName = log_dir + os.sep + "launcherLog.log"
     formatting = "%(asctime)s:%(levelname)s:%(name)s:%(message)s"
-    logging.basicConfig(
-        level=logging.DEBUG, filename=logFileName, filemode="w", format=formatting
-    )
+    # create file handler which logs even debug messages
+    fh = logging.FileHandler(logFileName, mode="w")
+    fh.setLevel(logging.DEBUG)
+    fh.setFormatter(logging.Formatter(formatting))
+    logger.addHandler(fh)
```
Comment (banesullivan) [inline python/src/wslink/backends/aiohttp/launcher.py:222]:
```suggestion
    logging.getLogger("wslink").addHandler(fh)
```

**Comment by banesullivan** [comment]
Not sure how to run the python tests in this package... someone should validate

File: python/src/wslink/backends/aiohttp/__init__.py [hunk 3]
```diff
@@ -163,7 +165,7 @@ async def stop(self):
 
 def create_webserver(server_config):
     if "logging_level" in server_config and server_config["logging_level"]:
-        logging.basicConfig(level=server_config["logging_level"])
+        logging.getLogger("wslink").setLevel(server_config["logging_level"])
```
Comment (jourdain) [inline python/src/wslink/backends/aiohttp/__init__.py:168]: should this be just `logger.setLevel(...)`

Comment (jourdain) [reply python/src/wslink/backends/aiohttp/__init__.py:168]: Actually, you are right... You want to drive the full hierarchy.

File: python/src/wslink/protocol.py [hunk 4]
```diff
@@ -252,7 +254,7 @@ async def onMessage(self, is_binary, msg, client_id):
             payload = payload.decode("utf-8")
 
         rpc = json.loads(payload)
-        logging.debug("wslink incoming msg %s" % self.payloadWithSecretStripped(rpc))
+        logger.debug("wslink incoming msg %s" % self.payloadWithSecretStripped(rpc))
```
Comment (jourdain) [inline python/src/wslink/protocol.py:257]: We may want to fix that line (while we are at it) to prevent string construction when the logger is disabled.

Comment (jourdain) [reply python/src/wslink/protocol.py:257]: Could you fix that with the following?

`logger.debug("wslink incoming msg %s", self.payloadWithSecretStripped(rpc))`

Comment (banesullivan) [reply python/src/wslink/protocol.py:257]:
```suggestion
        logger.debug("wslink incoming msg %s", self.payloadWithSecretStripped(rpc))
```

**Pull request merged**